File: Linear Data Structures/assignments/Java/Queue.java
/*
 * This is a Queue Data Structure Implementation Using LinkedList
 */
public class Queue {
    static class Node {
        Object value;
        Node next;

        Node (Object value) {
            this.value = value;
        }
    }

    private Node head;
    private Node tail;

    // Constructor
    /*
     * Utilize the tail pointer to achieve O(1) time complexity
     */
    public Queue() {
        this.head = null;
        this.tail = null;
    }

    /*
     * Inserting a value into the queue
     */
    public void enqueue(Object n) {
        // Code here
    }

    /* 
     * Removing a value from the queue using FIFO
     */
    public void dequeue() {
        // Code here
    }

    /*
     * Check if the Queue is empty
     * @return True if the queue is empty, else false
     */
    public boolean isEmpty() {
        return true; // Just a temporary placeholder
    }

    @Override
    /*
     * Printing out all the value in the queue
     */
    public String toString() {
        return "";
    }
}

File: Linear Data Structures/assignments/Python/Queue.py
class Node:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next

"""
Note: We will be using LinkedList for Queue Implementation.
"""
class Queue:
    # Note: Utilize the tail pointer to achieve O(1) enqueue
    # time complexity
    def __init__(self, head=None, tail=None):
        self.head = head
        self.tail = tail

    # Insert a value into a queue
    def enqueue(self, n):
        pass

    # Remove a value from a queue
    # If queue is empty return any message
    def dequeue(self):
        pass

    # Check if queue is empty
    def isEmpty(self):
        pass

    # Output all of the value(s) in queue
    def __str__(self):
        return
